content/learning-paths/smartphones-and-mobile/ray_tracing/rt07_shadows.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Shadows"
+weight: 8
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## Shadows
+
+Light sources emit light that will illuminate most objects. However, not all the emitted light will reach all objects; some light rays will be intercepted by other objects before they can reach a light source. This process creates darker areas and produces some shadows in our images.
+
+{{< tabpane >}}
+  {{< tab header="Example 1: ON" title="Example 1: shadows ON" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/city_shadows_on.png">}} {{< /tab >}}
+  {{< tab header="Example 1: OFF" title="Example 1: shadows OFF" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/city_shadows_off.png">}} {{< /tab >}}
+  {{< tab header="Example 2: ON" title="Example 2: shadows ON" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/bonza_shadows_on.png">}} {{< /tab >}}
+  {{< tab header="Example 2: OFF" title="Example 2: shadows OFF" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/bonza_shadows_off.png">}} {{< /tab >}}
+  {{< tab header="Example 3: ON" title="Example 3: shadows ON" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/immortalis_shadows_on.png">}} {{< /tab >}}
+  {{< tab header="Example 3: OFF" title="Example 3: shadows OFF" img_src="/learning-paths/smartphones-and-mobile/ray_tracing/images/immortalis_shadows_off.png">}} {{< /tab >}}
+{{< /tabpane >}}
+
+In our demo, we use a similar algorithm, to produce all our ray tracing effects. For shadows, similarly to what we did for reflections, we start by sampling some information from the G-buffer, like our position and normal. Then we use this information to generate shadows rays, that we can trace either using Ray Tracing Pipeline or Ray Query. If our shadow ray hits an object, it means that the ray is obstructed and we are in shadow, however, if the ray reaches the light without hitting anything it means that we are illuminated.
+
+![Diagram of our shadow algorithm #  center](images/shadows_algorithm_diagram.drawio.svg "Diagram of our shadow algorithm")
+
+When tracing shadows we launch a ray from the objects towards the lights, so we will need to launch a separate ray for each light that casts shadows, so please, consider limiting the number of lights casting shadows in your scenes. Launching a single ray per light source will only allow us to produce hard shadows, it is possible to obtain accurate soft shadows by launching multiple shadow rays and doing disk sampling. Soft shadows are more costly to produce, so please evaluate if they are necessary.
```

Comment:
disk sampling: might be good to have an external link here to an explanation?
also, there should be a "but" after the comma in that sentence ("hard shadows, but it is possible to...")
also, you're very polite with all the pleases! Maybe unnecessary? "so consider ..." "so evaluate..."?

Reply:
I think area sampling is more generic term. I added a link to a good tutorial.